jgiven-junit/src/test/java/com/tngtech/jgiven/junit/ScenarioExecutionTest.java

```diff
@@ -352,4 +352,17 @@ public void abstract_steps_should_appear_in_the_report_model() throws Throwable
 
     }
 
+    static class StageWithMissingState {
+        @ScenarioState( required = true )
```

Comment:
In particular, there should be a test with multiple steps where the second step sets the value.